src/tensors/cpu/prod.cpp

```diff
@@ -10,11 +10,9 @@
 #if MKL_FOUND
 #include <mkl.h>
 #elif BLAS_FOUND
```

Comment:
We only need SGEMM here?

Reply:
Almost everything here is handled in `prod_blas.h`, then there's the MKL batched gemm.

Reply:
We can get rid of the includes here, leaving only
```cpp
#if MKL_FOUND
#include <mkl.h>
#endif
```
since we directly call MKL functions here.

Agreed?